lib/block_view_lockups.go

```diff
@@ -739,16 +1075,36 @@ func (bav *UtxoView) _connectCoinLockup(
 			errors.Wrap(RuleErrorCoinLockupInvalidLockupDuration, "_connectCoinLockup")
 	}
 
-	// Compute the lockup duration in nanoseconds.
-	lockupDurationNanoSeconds := txMeta.UnlockTimestampNanoSecs - blockTimestampNanoSecs
+	// Validate the vesting end timestamp as expiring at a logically valid time.
+	if txMeta.VestingEndTimestampNanoSecs < txMeta.UnlockTimestampNanoSecs {
+		return 0, 0, nil,
+			errors.Wrap(RuleErrorCoinLockupInvalidVestingEndTimestamp, "_connectCoinLockup")
+	}
 
-	// Determine the hodler PKID to use.
-	transactorPKIDEntry := bav.GetPKIDForPublicKey(txn.PublicKey)
-	if transactorPKIDEntry == nil || transactorPKIDEntry.isDeleted {
+	// In the vested case, validate that the underlying profile is the transactor.
+	if txMeta.VestingEndTimestampNanoSecs > txMeta.UnlockTimestampNanoSecs &&
```

Comment:
Hmm... it looks like we're restricting vested lockups to only being creatable by the profile owner. Is this a necessary requirement? It kinda felt like any key should be able to send you locked coins of another person's supply. Example:
- N has minted 100 coins and sent them to A and B
- A wants to send B vested locked coins, which seems like it wouldn't be allowed here

I think this is also useful in the case where we want to isolate the minter of the coins from the distributor, which I think we probably want to do for security reasons.